backend/src/test/java/de/neuefische/CapStone/backend/controller/AuthControllerTest.java

```diff
@@ -61,5 +71,122 @@ public void createNewUser() {
+        assertThat(response.getStatusCode(), is(HttpStatus.CONFLICT));
+    }
+
+    @Test
+    public void userLogsIn() {
+        //GIVEN
+        String encodedPassword= passwordEncoder.encode("password");
+        String userName = "Bowser";
+        String email = "Bowser@supermario.com";
+        String role = "user";
+        String password = "password";
+        userRepository.save(
+                UserEntity.builder()
+                        .userName(userName)
+                        .email(email)
+                        .role(role)
+                        .password(encodedPassword).build()
+        );
+
+        Credentials credentials = Credentials.builder()
+                .userName(userName)
+                .password(password).build();
+        String newUrl = url() + "/user/login";
+
+        //WHEN
+      //  HttpEntity<Credentials> httpEntity = new HttpEntity<>(credentials);
+        ResponseEntity<AccessToken> response = testRestTemplate.postForEntity(newUrl, credentials, AccessToken.class);
+
+        //THEN
+        assertThat(response.getStatusCode(), is(HttpStatus.OK));
+        String token = response.getBody().getToken();
+        Claims claims = Jwts.parser().setSigningKey(jwtConfig.getSecret()).parseClaimsJws(token).getBody();
+        assertThat(claims.getSubject(), is(userName));
+        assertThat(claims.get("email", String.class), is(email));
+
+    }
+    @Test
+    public void unregisteredUserTriesToLogin() {
+        //GIVEN
+
+        Credentials credentials = Credentials.builder()
+                .userName("userName")
+                .password("password").build();
+        String newUrl = url() + "/user/login";
+
+        //WHEN
+      //  HttpEntity<Credentials> httpEntity = new HttpEntity<>(credentials);
+        ResponseEntity<AccessToken> response = testRestTemplate.postForEntity(newUrl, credentials, AccessToken.class);
+
+        //THEN
+        assertThat(response.getStatusCode(), is(HttpStatus.UNAUTHORIZED));
+    }
+
+    @Test
+    public void userTriesToLoginWithWrongPassword() {
+        //GIVEN
+        String encodedPassword= passwordEncoder.encode("password");
+        String userName = "Bowser";
+        String email = "Bowser@supermario.com";
+        String role = "user";
+        String password = "wrongpassword";
+        userRepository.save(
+                UserEntity.builder()
+                        .userName(userName)
+                        .email(email)
+                        .role(role)
+                        .password(encodedPassword).build()
+        );
+
+        Credentials credentials = Credentials.builder()
+                .userName(userName)
+                .password(password).build();
+        String newUrl = url() + "/user/login";
+
+        //WHEN
+        //  HttpEntity<Credentials> httpEntity = new HttpEntity<>(credentials);
+        ResponseEntity<AccessToken> response = testRestTemplate.postForEntity(newUrl, credentials, AccessToken.class);
+
```

Comment:
du könntest noch testen ob der request body null ist oder username oder passwort is null => sonst 500Server error